### PR TITLE
added the if statement

### PR DESCRIPTION
**Pull request by Hana3706**
solved the issue #5297
add a if statement

### CODE REVIEW

**Comment by llxia** [comment]
Please update the description to use https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by Hana3706** [comment]
testenvSettings.sh should run only once 

All four commits do the same thing, was a git mistake on my side 

Fixed #5297 and #5341

signed-off-by: Hana hanazayed06@gmail.com

**Comment by llxia** [comment]
Thanks @Hana3706 for the update. However, the code does not match the suggested https://github.com/adoptium/aqa-tests/pull/5341#discussion_r1608685076
Also, please do not include other changes that are unrelated to this issue. Thanks

**Comment by Hana3706** [comment]
@llxia am so sorry for this but can you point out to me how it doesn't match the code bcs on my local computer the code is updated to match the suggested changes, perhaps there was an issue with the version I commit...

**Comment by LongyuZhang** [comment]
Hi @Hana3706 Usually you can `rebase` to keep the PR up-to-date with the upstream repo.  For your current case, it might be easier to create a new branch with the following steps:
- make a copy of current branch (`git checkout -b <new_branch_1>`)
- checkout your master branch, fetch upstream, then rebase it to the latest
- Delete your current branch `my_new_branch` locally, then re-create new `my_new_branch` from master
- Make the changes (you can refer previous copied branch <new_branch_1>), commit and force push, then this PR will be updated.

**Comment by Hana3706** [comment]
@LongyuZhang thanks a lot for the feedback! 
I was just wondering why I need to create a new branch? As well as that, I want to better understand what is wrong with the update I made so I can fix it again.
Finally, how do I rebase to the latest branch upstream and how do I delete a branch locally...
am so sorry for the many questions :)
thanks again!

**Comment by LongyuZhang** [comment]
@Hana3706 If you can rebase on the currently branch, it would be great. I suggested re-create new branch since you commits may already mixed with merged commits. So feel free to try rebase on this branch first.
Regarding the rebase, you can follow https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-repository-for-a-fork to add `adoptium/aqa-tests` as the `upstream`, `fetch` it to your local, then use `git rebase upstream/master` command to update your branch.

File: buildenv/jenkins/JenkinsfileBase [hunk 1]
```diff
@@ -6,7 +6,10 @@ import org.tap4j.model.TestSet;
 def makeTest(testParam) {
 	String tearDownCmd = "$RESOLVED_MAKE; \$MAKE -f ./aqa-tests/TKG/testEnv.mk testEnvTeardown"
 	// Note: keyword source cannot be used in Jenkins script. Therefore, using "." instead.
-	String makeTestCmd = "$RESOLVED_MAKE;cd ./aqa-tests; . ./scripts/testenv/testenvSettings.sh;cd ./TKG; \$MAKE $testParam"
+	String makeTestCmd = "$RESOLVED_MAKE";cd. /aqa-tests/TKG; \$MAKE $testParam"
```
Comment (LongyuZhang) [inline buildenv/jenkins/JenkinsfileBase:9]: the `cd.` should be `cd .`

File: buildenv/jenkins/JenkinsfileBase [hunk 2]
```diff
@@ -6,7 +6,10 @@ import org.tap4j.model.TestSet;
 def makeTest(testParam) {
 	String tearDownCmd = "$RESOLVED_MAKE; \$MAKE -f ./aqa-tests/TKG/testEnv.mk testEnvTeardown"
 	// Note: keyword source cannot be used in Jenkins script. Therefore, using "." instead.
```
Comment (LongyuZhang) [inline buildenv/jenkins/JenkinsfileBase:8]: This comment should be moved into the `if` block too

Comment (Hana3706) [reply buildenv/jenkins/JenkinsfileBase:8]: To make sure, do you want me to move the import otf.tap4j.model.TestSet comment inside my if statement on line 10 or before it on line 9?

Comment (LongyuZhang) [reply buildenv/jenkins/JenkinsfileBase:8]: This comment `// Note: keyword source cannot be used in Jenkins script. Therefore, using "." instead.` is for `makeTestCmd = "$RESOLVED_MAKE;cd ./aqa-tests; . ./scripts/testenv/testenvSettings.sh;cd ./TKG; \$MAKE $testParam"`, so they should be together.

File: buildenv/jenkins/JenkinsfileBase [hunk 3]
```diff
@@ -6,7 +6,10 @@ import org.tap4j.model.TestSet;
...
-	String makeTestCmd = "$RESOLVED_MAKE;cd ./aqa-tests; . ./scripts/testenv/testenvSettings.sh;cd ./TKG; \$MAKE $testParam"
+	String makeTestCmd = "$RESOLVED_MAKE";cd. /aqa-tests/TKG; \$MAKE $testParam"
+	if (env.DYNAMIC_COMPILE){
+		makeTestCmd = "$RESOLVED_MAKE;cd ./aqa-tests; . ./scripts/testenv/testenvSettings.sh;cd ./TKG; \$MAKE $testParam"
+	}
```
Comment (smlambert) [inline buildenv/jenkins/JenkinsfileBase:12]:
```suggestion
	if (env.DYNAMIC_COMPILE){
		// Note: keyword source cannot be used in Jenkins script. Therefore, using "." instead.
		makeTestCmd = "$RESOLVED_MAKE;cd ./aqa-tests; . ./scripts/testenv/testenvSettings.sh;cd ./TKG; \$MAKE $testParam"
	} 
```

File: buildenv/jenkins/JenkinsfileBase (hunk 1 above)
Comment (smlambert) [inline buildenv/jenkins/JenkinsfileBase:9]:
```suggestion
	String makeTestCmd = "$RESOLVED_MAKE";cd ./aqa-tests/TKG; \$MAKE $testParam"
```

**Review by smlambert — CHANGES_REQUESTED** [review]

**Comment by llxia** [comment]
@Hana3706 are you still working on this PR? If so, please address the above comments. Thanks

**Comment by smlambert** [comment]
Closing as stale.  Reopen if requested updates are made.

**Pull request closed without merging**